app/main.py

```diff
@@ -1 +1,32 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return ("{Name: " + self.name + ", Health: " + str(self.health)
+                + ", Hidden: " + str(self.hidden) + "}")
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    @staticmethod
+    def bite(other: Herbivore) -> None:
+        if not isinstance(other, Carnivore) and not other.hidden:
+            if other.health - 50 < 1:
+                Animal.alive.remove(other)
+            else:
+                other.health -= 50
```

Comment:
you could a little simplify this by removing health at first, and make check at second.